examples/notebooks/beam-ml/rag_usecase/opensearch_enrichment.py

```diff
@@ -0,0 +1,130 @@
+the similar text along with its embeddings as Beam.Row Object.
+
+Example usage::
+  opensearch_handler = opensearchEnrichmentHandler(opensearch_host='127.0.0.1', opensearch_port=6379)
+
+  pipeline | Enrichment(opensearch_handler)
+
+No backward compatibility guarantees. Everything in this module is experimental.
+"""
+
+import logging
+
+
+from opensearchpy import OpenSearch
+from typing import Optional
+import os
+
+import apache_beam as beam
+from apache_beam.transforms.enrichment import EnrichmentSourceHandler
+from apache_beam.transforms.enrichment_handlers.utils import ExceptionLevel
+
+__all__ = [
+    'OpenSearchEnrichmentHandler',
+]
+
+# Set the logging level to reduce verbose information
+import logging
+
+logging.root.setLevel(logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+class OpenSearchEnrichmentHandler(EnrichmentSourceHandler[beam.Row, beam.Row]):
+    """A handler for :class:`apache_beam.transforms.enrichment.Enrichment`
+    transform to interact with opensearch vector DB.
+    """
+    def __init__(
+            self,
+            opensearch_host: str,
+            opensearch_port: int,
+            username: Optional[str] = os.getenv("OPENSEARCH_USERNAME"),
+            password: Optional[str] = os.getenv("OPENSEARCH_PASSWORD"),
+            index_name: str = "embeddings-index",
+            vector_field: str = "text_vector",
+            k: int = 1,
+    ):
+
+        """Args:
+          opensearch_host (str): opensearch Host to connect to opensearch DB
+          opensearch_port (int): opensearch Port to connect to opensearch DB
+          index_name (str): Index Name created for searching in opensearch DB
+          vector_field (str): vector field to compute similarity score in vector DB
+          k (int): Value of K in KNN algorithm for searching in opensearch
+        """
+        self.opensearch_host = opensearch_host
+        self.opensearch_port = opensearch_port
+        self.username = username
+        self.password = password
+        self.index_name = index_name
+        self.vector_field = vector_field
+        self.k = k
+        self.client = None
+
+    def __enter__(self):
+        """connect to the opensearch DB using opensearch client.
+        """
+
+        if self.client is None:
+            http_auth = [self.username, self.password]
+            self.client = OpenSearch(hosts=[f'{self.opensearch_host}:{self.opensearch_port}'],
+                                     http_auth=http_auth,
+                                     verify_certs=False)
+
+    def __call__(self, request: beam.Row, *args, **kwargs):
+        """
+        Reads a row from the opensearch Vector DB and returns
+        a `Tuple` of request and response.
+
+        Args:
+        request: the input `beam.Row` to enrich.
+        """
+
+        # read embedding vector for user query
+
+        embedded_query = request['text']
+
+        # Prepare the Query
+        query = {
+            'size': 5,
```

Comment:
Should this be a const or a configurable parameter?

Reply:
Done